src/main/java/core/basesyntax/service/strategy/handler/BalanceActivityHandler.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.service.strategy.handler;
+
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.FruitTransaction;
+
+public class BalanceActivityHandler implements ActivityHandler {
+    private final FruitDao fruitDao;
+
+    public BalanceActivityHandler(FruitDao fruitDao) {
+        this.fruitDao = fruitDao;
+    }
+
+    @Override
+    public void accept(FruitTransaction fruitTransaction) {
+        Fruit fruit = fruitTransaction.getFruit();
+        if (fruitTransaction.getQuantity() < 0) {
```

Comment:
this check better perform when FruitTransaction is creating (in `getFruitTransactionsFromData(List<String> activities);`)
here you should have a valid object.